addons/website_slides_survey/tests/test_course_certification_unlink.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo.tests import tagged
+from odoo.tests.common import  TransactionCase
+from odoo.exceptions import ValidationError, UserError
+
+
+@tagged('unlink')
+class TestCourseCertifUnlink(TransactionCase):
+
+    def test_unlink(self):
+        self.survey = self.env['survey.survey'].create({
+            'title': 'A new Certification',
+        })
+
+        self.course = self.env['slide.channel'].sudo().create({'name': 'Testing Channel'})
+        self.certification = self.env['slide.slide'].sudo().create({
+            'name': 'Certification',
+            'slide_type': 'certification',
+            'channel_id': self.course.id,
+            'survey_id': self.survey.id,
+            'user_id': 1,
+        })
+        self.assertEqual(len(self.course.slide_ids), 1)
+
+        self.executed = False
+        with self.assertRaises(ValidationError, msg="The error is unexpected"):
+            self.executed = self.survey.unlink()
+
+        self.assertFalse(self.executed)
+        self.certification['survey_id'] = None
+        try:
+            self.executed = self.survey.unlink()
+        except (ValidationError, UserError):
+            self.assertTrue(1 == 0, "An exception occurred")
+        self.assertTrue(self.executed)
```

Comment:
This code is a bit weird, you can do
 ```suggestion
        self.certification.survey_id = False
        self.survey.unlink()
        self.assertFalse(self.survey.exists())
```

It's ok if the exception can be raised during the test as it will be explicit